src/utils/downloadArtifact.ts

```diff
@@ -33,3 +35,36 @@ export async function downloadArtifact(artifact, destFolder) {
   await zip.extract(null, destFolder);
   await zip.close();
 }
+
+export async function downloadSameWorkflowArtifacts() {
+  const client = create();
+  // Try to download all artifacts from the current workflow, but do not fail the build if this fails
+  const artifacts = await client.downloadAllArtifacts(cacheDir).catch((e) => {
```

Comment:
Where do you ensure that these artifacts are belongs to the current workflow?

Reply:
That's this method: https://github.com/actions/toolkit/blob/e6257f111756d2f3567917c8e27ab57de8c3e09c/packages/artifact/src/internal/artifact-client.ts#L221
using this method to list artifacts: https://github.com/actions/toolkit/blob/e6257f111756d2f3567917c8e27ab57de8c3e09c/packages/artifact/src/internal/download-http-client.ts#L45
which in turn is using https://github.com/actions/toolkit/blob/e6257f111756d2f3567917c8e27ab57de8c3e09c/packages/artifact/src/internal/utils.ts#L222 as the download URL
which is referencing the current workflow run using the env variable here:
https://github.com/actions/toolkit/blob/e6257f111756d2f3567917c8e27ab57de8c3e09c/packages/artifact/src/internal/config-variables.ts#L50

> Looks like you are downloading all the files, which may contain hundreds of gigs in my case.

Good point. An alternative would be using `@actions/artifact`s internals as I attempted here:
[zwave-js/node-zwave-js@`10121ae` (#5050)](https://github.com/zwave-js/node-zwave-js/pull/5050/commits/10121ae59bc706d2cee6845adfde50c244e738cc#diff-092e3ce93a34bddd9c7d3fcf473051fd3d4aae325c76015ff004a2071f6df7e9L60-L85)
This way one could filter for files that look like a hash, but that might not be stable across releases of that package.

Reply:
If you don't like this approach, I first attempted to download the files from the current workflow and fall back to the ones from other workflows if that didn't work. However, it was pretty slow (20 seconds for a handful of small artifacts).